### ISSUE TITLE
Add sample that uses `Selector.AddSend`

### ISSUE THREAD

**Issue by cretz** [temporalio/samples-go#142]
The docs currently say:

> // The sending message to the channel is expected to be done by the callback function

But this is unclear.

**Comment by ioannist** [comment]
Indeed, the comments of the Selector interface are not very helpful and there are no examples that show how the different triggers differ and when they should be used.

https://github.com/temporalio/sdk-go/blob/master/internal/workflow.go
